pkg/skaffold/deploy/helm/helm_test.go

```diff
@@ -245,21 +245,37 @@ var testDeployWithTemplatedName = latest.HelmDeploy{
 	},
 }
 
-var testDeploySkipBuildDependencies = latest.HelmDeploy{
+var testDeployRepositoryChartSkipBuildDependencies = latest.HelmDeploy{
+	Repositories: []latest.HelmRepository{{
+		Name: "stable",
+		URL:  "https://charts.helm.sh/stable",
+	}},
 	Releases: []latest.HelmRelease{{
-		Name:      "skaffold-helm",
-		ChartPath: "stable/chartmuseum",
-		ArtifactOverrides: map[string]string{
-			"image.tag": "skaffold-helm",
-		},
+		Name:                  "skaffold-helm",
+		Repository:            "stable",
+		ChartPath:             "chartmuseum",
 		SkipBuildDependencies: true,
 	}},
 }
 
-var testDeployRemoteChart = latest.HelmDeploy{
```

Comment:
I having some trouble with the specs. Its reporting the warning that the image tag is not being tracked by Skaffold.

I don't believe this test case was correct as its attempting to test a remote chart but the Remote field is not set to true. When doing this on master I get the same result now faced.

```
➜  skaffold git:(35214eb81) hack/gotest.sh  -count=1 -race -short -timeout=90s ./pkg/skaffold/deploy/helm/...                                                                                                     
go test -count=1 -race -short -timeout=90s ./pkg/skaffold/deploy/helm/...
FAIL    github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/helm       0.335s

=== Failed Tests ===
/pkg/skaffold/deploy/helm/TestHelmDeploy
--- FAIL: TestHelmDeploy (0.09s)
/pkg/skaffold/deploy/helm/TestHelmDeploy/deploy_success_remote_chart_with_skipBuildDependencies
time="2020-12-28T08:18:39-08:00" level=info msg="Deploying with helm v3.1.1 ..."
time="2020-12-28T08:18:39-08:00" level=info msg="Release skaffold-helm not upgraded as it is remote..."
    TestHelmDeploy/deploy_success_remote_chart_with_skipBuildDependencies: helm_test.go:973: []string differ (-got, +want):   []string(                                                                           
        -       {
        -               "See helm sample for how to replace image names with their actual tags: https://github.com/GoogleContainerTools/skaffold/blob/mas"...,                                                    
        -               "image [docker.io:5000/skaffold-helm:3605e7bc17cf46e53f4d81c4cbc24e5b4c495184] is not used.",                                                                                             
        -               "image [skaffold-helm] is used instead.",
        -       },
        +       nil,
          )
    --- FAIL: TestHelmDeploy/deploy_success_remote_chart_with_skipBuildDependencies (0.00s)
/pkg/skaffold/deploy/helm/TestHelmDeploy/deploy_error_remote_chart_without_skipBuildDependencies
time="2020-12-28T08:18:39-08:00" level=info msg="Deploying with helm v3.1.1 ..."
time="2020-12-28T08:18:39-08:00" level=info msg="Release skaffold-helm-remote not upgraded as it is remote..."                                                                                                    
    TestHelmDeploy/deploy_error_remote_chart_without_skipBuildDependencies: helm_test.go:972: expected error, but returned none                                                                                   
    TestHelmDeploy/deploy_error_remote_chart_without_skipBuildDependencies: helm_test.go:973: []string differ (-got, +want):   []string(                                                                          
        -       {
        -               "See helm sample for how to replace image names with their actual tags: https://github.com/GoogleContainerTools/skaffold/blob/mas"...,                                                    
        -               "image [docker.io:5000/skaffold-helm:3605e7bc17cf46e53f4d81c4cbc24e5b4c495184] is not used.",                                                                                             
        -               "image [skaffold-helm] is used instead.",
        -       },
        +       nil,
          )
    TestHelmDeploy/deploy_error_remote_chart_without_skipBuildDependencies: helm_test.go:974: expected error, but returned none                                                                                   
    --- FAIL: TestHelmDeploy/deploy_error_remote_chart_without_skipBuildDependencies (0.00s)

=== Slow Tests ===
0.09s   TestHelmDeploy
0.02s   TestHelmDeploy/helm3.0_namespaced_(with_env_template)_deploy_success
0.01s   TestHelmDeploy/helm3.1_namespaced_deploy_(with_env_template)_success
0.01s   TestParseHelmRelease
0.01s   TestHelmRender
0.01s   TestParseReleaseInfo/parse_valid_release_info_yaml_with_multiple_artifacts
0.01s   TestParseReleaseInfo
exit status 1
```

The new specs should cover running but die after the call to `helm get skaffold-helm`. I believe this if from the upgrade not occurring when a remote repository is used.

I'd appreciate some help here to get the correct functionality covered by specs.

```
helm repo add
helm install 
```

The changes do work as I've built the project and booted up the project that originally got my eyes on this feature enhancement.